source/Halibut/Transport/TcpClientExtensions.cs

```diff
@@ -33,8 +37,31 @@ static async Task Connect(TcpClient client, string host, int port, TimeSpan time
         {
             try
             {
-                var task = client.ConnectAsync(host, port);
-                await task.TimeoutAfter(timeout, cancellationToken);
+                var elapsedDuration = Stopwatch.StartNew();
+                var hostAddressesTask = Dns.GetHostAddressesAsync(host);
```

Comment:
This is a less desirable bit of complexity. When keep alive are enabled on linux then you can't connect via host but need to resolve the ip address first and then connect